### PR TITLE
Replace endpoint strings by NodeLocator objects

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Legacy endpoints "host:port" have replaced by NodeLocator objects.

Why should this be merged: 
It's impossible to use NodeLocator as endpoints in tests because of legacy methods in tests which uses old "host:port" format.

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 1]
```diff
@@ -275,6 +284,7 @@
      * Initiates the failure detection and healing detection tasks.
      */
     private void initializationTask() {
+        log.info("Initialization task ...");
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: I would add a little bit more context to this info as in initialization of what task?

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 2]
```diff
@@ -390,7 +403,7 @@ private SequencerMetrics queryLocalSequencerMetrics(Layout layout) {
         SequencerMetrics sequencerMetrics;
         // This is an optimization. If this node is not the primary sequencer for the current
         // layout, there is no reason to request metrics from this sequencer.
-        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
+        if (layout.getSequencersNodes().get(0).equals(getLocalEndpoint())) {
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: should be either getSequencers or getSequencerNodes

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 3]
```diff
@@ -457,13 +471,12 @@ private synchronized void detectorTaskScheduler() {
...
-
         // We check for the following condition here: If the node is NOT a part of the
         // current layout, it should not attempt to change layout.
+
         Layout layout = serverContext.getManagementLayout();
-        if (!layout.getAllServers().contains(getLocalEndpoint())) {
-            log.debug("This Server is not a part of the active layout. "
-                    + "Aborting reconfiguration handling.");
+
+        if (!layout.getAllServersNodes().contains(getLocalEndpoint())) {
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: it should be getAllServers or getAllServerNodes

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 4]
```diff
@@ -457,13 +471,12 @@ private synchronized void detectorTaskScheduler() {
...
             return false;
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: I would flip the logic.. 
******
if(layout.getAllServerNodes().contains(getLocakEndpoint())
   return true;

return false;
******

If this is a cluster of 3 and if cluster is stable then you can be more efficient by checking for a state that is more prevalent.

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 5]
```diff
@@ -514,17 +527,18 @@ private void runHealingDetectorTask() {
...
+                                .getLayoutView()
+                                .getRuntimeLayout(layout)
                                 .getManagementClient(getLocalEndpoint())
-                                .handleHealing(pollReport.getPollEpoch(),
-                                        pollReport.getHealingNodes())
+                                .handleHealing(pollReport.getPollEpoch(), pollReport.getHealingNodes())
                                 .get();
                         log.info("Healing nodes successful: {}", pollReport);
                     } catch (InterruptedException | ExecutionException e) {
+                        Thread.currentThread().interrupt();
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: good practice to set the interrupt flag of the thread

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java [hunk 6]
```diff
@@ -709,8 +729,10 @@ private void correctOutOfPhaseEpochs(PollReport pollReport) {
             // Retrieve the correct layout from quorum of members to reseal servers.
             // If we are unable to reach a consensus from a quorum we get an exception and
             // abort the epoch correction phase.
-            Layout quorumLayout = fetchQuorumLayout(layoutCompletableFutureMap.values()
-                    .toArray(new CompletableFuture[layoutCompletableFutureMap.size()]));
+            CompletableFuture[] layoutFutures = layoutCompletableFutureMap
+                    .values()
+                    .toArray(new CompletableFuture[layoutCompletableFutureMap.size()]);
+            Layout quorumLayout = fetchQuorumLayout(layoutFutures);
```
Comment (apavan) [inline infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: If this is only for better readability then we could also have formatted the code with proper indentation.

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 2 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: `getSequencers()` marked as deprecated, will be deleted in future versions

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 1 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 3 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: fixed

Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: I mean now, alomost everywhere we use getAllServerNodes

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 4 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: fixed

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 5 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: Yep, this is exactly what I did

File: infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java (hunk 6 above)
Comment (xnull) [reply infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java]: It is a proper indentation. Formatted by intellij idea

**Pull request closed without merging**